internal/graph/check.go

```diff
@@ -173,8 +178,31 @@ func (cc *ConcurrentChecker) checkInternal(ctx context.Context, req ValidatedChe
 		return checkResultsForMembership(membershipSet, emptyMetadata)
 	}
 
+	// Filter for check hints, if any.
+	if len(req.CheckHints) > 0 {
+		filteredResourcesIdsSet := mapz.NewSet(filteredResourcesIds...)
+		for _, checkHint := range req.CheckHints {
+			resourceID, ok := hints.AsCheckHintForComputedUserset(checkHint, req.ResourceRelation, req.Subject)
```

Comment:
> why can we assume here it's a hint for computed usersets?

Because this is processing for the relation itself, and not the arrow.

> You could return the result from the hint along side the resourceID...

Except when there aren't any hints, then we're doing work we don't need to do. Its also capping at maybe 1000 elements right now, which means the overhead is likely minimal. I can combine if you like, but it makes the code less readable for a very small improvement

Reply:
>Except when there aren't any hints, then we're doing work we don't need to do.

What work, creating an empty map?

>Its also capping at maybe 1000 elements right now, which means the overhead is likely minimal. I can combine if you like, but it makes the code less readable for a very small improvement

How is it making it less readable? You already have a method that accepts a map.
Looking at these things as 1 call with 1000 elements wouldn't be much overhead. The issue is a graph with many elements can lead to many, many dispatches. I think every little bit counts.